pkg/clients/default-domain/client_test.go

```diff
@@ -0,0 +1,129 @@
+package defaultdomain
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewClient(t *testing.T) {
+	opts := Opts{
+		ServerAddress: "http://localhost:9000",
+	}
+
+	client := NewClient(opts)
+
+	require.NotNil(t, client)
+	assert.Equal(t, opts, client.opts)
+	assert.NotNil(t, client.httpClient)
+}
+
+func TestClient_GetTLSCertificate_Success(t *testing.T) {
+	expiresOn := time.Now().Add(24 * time.Hour)
+	expectedCert := &TLSCertificate{
+		Key:       []byte("test-key"),
+		Cert:      []byte("test-cert"),
+		ExpiresOn: &expiresOn,
+	}
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, http.MethodGet, r.Method)
+		assert.Equal(t, "/defaultdomain/subscriptions/sub-123/resourcegroups/rg-test/clusters/cluster-1/ccpid/ccp-456/defaultdomaintls", r.URL.Path)
```

Comment:
ok I see how this is tested, I still think decomposition might be helpful but this is pretty clever